File: common/src/main/java/cc/turtl/turtlshell/command/TurtlShellCommand.java
package cc.turtl.turtlshell.command;

import com.mojang.brigadier.builder.LiteralArgumentBuilder;

public interface TurtlShellCommand {

    /**
     * The command name in lowercase (e.g., "info", "debug")
     */
    String getName();

    /**
     * Short description for help text
     */
    String getDescription();

    /**
     * Build and return the command structure.
     * Use Commands.literal(getName()) and attach logic.
     */
    <S> LiteralArgumentBuilder<S> build();
}

File: common/src/main/kotlin/cc/turtl/turtlshell/api/TurtlShellCommand.kt
package cc.turtl.turtlshell.api

import com.mojang.brigadier.builder.LiteralArgumentBuilder
import net.minecraft.commands.CommandSourceStack

interface TurtlShellCommand {
    /**
     * The command name in lowercase (e.g., "info", "debug")
     */
    val name: String

    /**
     * Short description for help text
     */
    val description: String get() = ""

    /**
     * Build and return the command structure.
     * Use Commands.literal(getName()) and attach logic.
     */
    fun build(): LiteralArgumentBuilder<CommandSourceStack>
}